merlin-server/sql/merlin/tables/constraint.sql

```diff
@@ -82,3 +82,20 @@ create trigger set_timestamp
 before update on "constraint"
 for each row
 execute function constraint_set_updated_at();
+
+create or replace function constraint_check_constraint_run()
+returns trigger
+security definer
+language plpgsql as $$begin
+  update constraint_run
+  set status = 'constraint-outdated'
+  where constraint_id = new.id
+  and constraint_definition != new.definition
+  and status = 'resolved';
+  return new;
+end$$;
+
+create trigger constraint_check_constraint_run_trigger
+after update on "constraint"
+for each row
```

Comment:
You can add the following condition on this trigger to avoid running it when the definition doesn't change:
```sql
when (new.definition != old.definition`)
```